File: backend/src/models/TaskTemplate.js
import mongoose from 'mongoose';

const taskTemplateSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  title: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    default: '',
  },
  tags: [{
    type: String,
  }],
  priority: {
    type: String,
    enum: ['low', 'medium', 'high'],
    default: 'medium',
  },
  estimatedHours: {
    type: Number,
    default: 0,
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
}, {
  timestamps: true,
});

export default mongoose.model('TaskTemplate', taskTemplateSchema);

File: backend/app/schemas/task_template.py
from pydantic import BaseModel
from typing import Optional, List


class TemplateCreate(BaseModel):
    name: str
    title: str
    description: Optional[str] = ""
    tags: Optional[List[str]] = []
    priority: Optional[str] = "medium"
    estimatedHours: Optional[float] = 0
